src/Compilers/CSharp/Portable/Symbols/TupleFieldSymbol.cs

```diff
@@ -0,0 +1,230 @@
+        ///
+        /// index is 1 for Item1.
+        /// </summary>
+        internal TupleFieldSymbol(string name, TupleTypeSymbol containingTuple, TypeSymbol type, int position, FieldSymbol underlyingField)
+        {
+            _name = name;
+            _containingTuple = containingTuple;
+            _underlyingFieldOpt = underlyingField;
+            _type = type;
+            _position = position;
+        }
+
+        /// <summary>
+        /// Copy this tuple field, but modify it to use the new containing tuple and underlying type.
+        /// </summary>
+        internal TupleFieldSymbol WithUnderlyingType(TupleTypeSymbol newContainingTuple, NamedTypeSymbol newContainingUnderlyingType)
+        {
+            if ((object)_underlyingFieldOpt == null)
+            {
+                return null;
+            }
+
+            FieldSymbol newUnderlyingField = _underlyingFieldOpt.OriginalDefinition.AsMember(newContainingUnderlyingType);
+            return new TupleFieldSymbol(_name, newContainingTuple, newUnderlyingField.Type, _position, newUnderlyingField);
+        }
+
+        internal static FieldSymbol GetUnderlyingField(int containingTupleArity, NamedTypeSymbol containingUnderlyingType, int fieldIndex, CSharpSyntaxNode syntax, Binder binder, DiagnosticBag diagnostics)
+        {
+            int tupleRemainder;
+            int tupleChainLength = TupleTypeSymbol.NumberOfValueTuples(containingTupleArity, out tupleRemainder);
+
+            int fieldRemainder;
+            int fieldChainLength = TupleTypeSymbol.NumberOfValueTuples(fieldIndex + 1, out fieldRemainder);
+
+            int containingLinkSize = 8;
+            if (fieldChainLength == tupleChainLength)
+            {
+                containingLinkSize = tupleRemainder;
+            }
+
+            // PROTOTYPE(tuples) Constructing a TupleTypeSymbol with TupleFieldSymbols should not produce diagnostics for missing well-known members and type members.
+            //                      The error handling should be shifted to lowering. This will help when we implement metadata loading.
+            WellKnownMember wellKnownTypeMember = TupleTypeSymbol.GetTupleTypeMember(containingLinkSize, fieldRemainder);
+            var linkField = (FieldSymbol)Binder.GetWellKnownTypeMember(binder.Compilation, wellKnownTypeMember, diagnostics, syntax: syntax);
+            if ((object)linkField == null)
+            {
+                return null;
+            }
+
+            NamedTypeSymbol linkType = TupleTypeSymbol.GetNestedTupleType(containingUnderlyingType, fieldChainLength - 1);
+
+            return linkField.AsMember(linkType);
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return _name;
+            }
+        }
+
+        /// <summary>
+        /// Returns the index of this field within the tuple.
+        /// For instance, the index for Item1 is 1.
```

Comment:
I didn't understand. I changed from index (zero-based) to a position (one-based).